### PR TITLE
add instrumented serivce  unit test of version and status

### PR DESCRIPTION
**Pull request by drinktee**
What this PR does / why we need it:
Increases test coverage for kubelet/kuberuntime
#46123

Which issue this PR fixes:
#46123

### CODE REVIEW

**Comment by drinktee** [comment]
ping  @yujuhong  thx..

File: pkg/kubelet/kuberuntime/instrumented_services_test.go [hunk 1]
```diff
@@ -59,3 +59,19 @@ func TestRecordOperation(t *testing.T) {
...
+	assert.NoError(t, err)
+	assert.Equal(t, kubeRuntimeAPIVersion, vr.Version)
+}
+
+func TestInstrumentedStatus(t *testing.T) {
+	fakeRuntime, _, _, _ := createTestRuntimeManager()
+	irs := newInstrumentedRuntimeService(fakeRuntime)
+	rs, err := irs.Status()
+	assert.NoError(t, err)
+	assert.Equal(t, "nil", rs.String())
```
Comment (yujuhong) [inline pkg/kubelet/kuberuntime/instrumented_services_test.go]: I'm not exactly sure what this test is verifying...why should the status be a nil string?

**Comment by drinktee** [comment]
Just to test the status method....The FakeRuntimeService will return a FakeStatus which is nil.

On 10/10/2017 23:58, Yu-Ju Hong wrote: @yujuhong commented on this pull request.



In pkg/kubelet/kuberuntime/instrumented_services_test.go:
> @@ -59,3 +59,19 @@ func TestRecordOperation(t *testing.T) {
 		mux.ServeHTTP(w, r)
 	}), "GET", prometheusUrl, nil, runtimeOperationsLatencyExpected)
 }
+
+func TestInstrumentedVersion(t *testing.T) {
+	fakeRuntime, _, _, _ := createTestRuntimeManager()
+	irs := newInstrumentedRuntimeService(fakeRuntime)
+	vr, err := irs.Version("1")
+	assert.NoError(t, err)
+	assert.Equal(t, kubeRuntimeAPIVersion, vr.Version)
+}
+
+func TestInstrumentedStatus(t *testing.T) {
+	fakeRuntime, _, _, _ := createTestRuntimeManager()
+	irs := newInstrumentedRuntimeService(fakeRuntime)
+	rs, err := irs.Status()
+	assert.NoError(t, err)
+	assert.Equal(t, "nil", rs.String())

I'm not exactly sure what this test is verifying...why should the status be a nil string?

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/kubernetes/kubernetes","title":"kubernetes/kubernetes","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/kubernetes/kubernetes"}},"updates":{"snippets":[{"icon":"PERSON","message":"@yujuhong commented on #53294"}],"action":{"name":"View Pull Request","url":"https://github.com/kubernetes/kubernetes/pull/53294#pullrequestreview-68334370"}}}

**Comment by yujuhong** [comment]
> Just to test the status method....The FakeRuntimeService will return a FakeStatus which is nil.

The point is....if you are testing this method, using some more meaningful data would make sense :-)

File: pkg/kubelet/kuberuntime/instrumented_services_test.go (hunk 1 above)
Comment (feiskyer) [reply pkg/kubelet/kuberuntime/instrumented_services_test.go]: agreed. better to set `FakeStatus` inside createTestRuntimeManager()

Comment (drinktee) [reply pkg/kubelet/kuberuntime/instrumented_services_test.go]: May I remove this test first? :-)

**Pull request closed without merging**